openapi.yaml

```diff
@@ -1023,6 +1055,74 @@ components:
                   description: Job reason code
                   example: ABCDE
                   nullable: true
+    DeductionsResource:
+      type: object
+      properties:
+        id:
+          type: string
+          description: OSU ID of a employee
```

Comment:
This is incorrect. Our APIs should also adhere to [JSON:API standard](https://jsonapi.org/format/#document-resource-object-identification) as well. Base on that, if your resource is an employee, then you have to redesign your endpoint to something like `/servicenow/:osuId/deductions`, otherwise, the ID here should be something else unique to that resource.

Reply:
Ok, I wasn't sure about this one since there isn't really one unique ID, we're creating 13 different payroll deductions for an employee. Is that ID property required?

Reply:
Yes, a unique resource ID is required based on JSON:API:

> Every resource object MUST also contain an id member, except when the resource object originates at the client and represents a new resource to be created on the server. If id is omitted due to this exception, a lid member MAY be included to uniquely identify the resource by type locally within the document.

You don't need to have it mapped with a real field in the database. As long as it's unique, you can use it as a deductions ID. For example, you can use `[osuid]-[position number]` as the resource ID.

Reply:
Got it, I'll come up with something for this. Thanks